src/pybind/mgr/volumes/fs/operations/versions/__init__.py

```diff
@@ -81,7 +81,7 @@ def upgrade_legacy_subvolume(self, fs, subvolume):
         subvolume.init_config(SubvolumeV1.version(), subvolume_type, qpath, initial_state)
 
     def get_subvolume_object(self, fs, vol_spec, group, subvolname, upgrade=True):
-        subvolume = SubvolumeBase(fs, vol_spec, group, subvolname)
+        subvolume = SubvolumeDiscover(fs, vol_spec, group, subvolname)
```

Comment:
Why carve this out into a separate class?

(also, SubvolumeDiscover does not have abstract discover method)

Reply:
> Why carve this out into a separate class?

The intention was to provide abstract methods in the base class, hence it renders creating an instance of the base class invalid, thus leading to a derived class for help with discovery.

> (also, SubvolumeDiscover does not have abstract discover method)

Do you mean we should move discover into this class and make it abstract in the base class? Which sounds like a good idea to me. (currently discover is not an abstract method)

Reply:
> Do you mean we should move discover into this class and make it abstract in the base class? Which sounds like a good idea to me. (currently discover is not an abstract method)

Right, I expected discover() as an abstract method. Thinking more -- can upgrade() be an abstact method that can be invoked to upgrade a subvolume to the latest version?

Reply:
Now that V2 is derived from V1, the need for an abstract base class is not present any longer. Hence SubvolumeDiscover is removed.

I did play around with an upgrade method briefly earlier, but it did not help much. Let's leave the upgrade explicit during the `get_subvolume_object` process.